pkg/bootstrapper/bootstrapper.go

```diff
@@ -0,0 +1,441 @@
+func prepKubeletConfForWindows(bs *winNodeBootstrapper, initialConfig []byte) ([]byte, error) {
+	var out []byte
+	// Here we parse the initial configuration, which was yaml, into a KubeletConfiguration struct
+	b := bufio.NewReader(bytes.NewReader(initialConfig))
+	r := yaml.NewYAMLReader(b)
+	doc, err := r.Read()
+	if err != nil {
+		return out, err
+	}
+	scheme := runtime.NewScheme()
+	err = kubeletConfig.AddToScheme(scheme)
+	if err != nil {
+		return out, err
+	}
+	d := serializer.NewCodecFactory(scheme).UniversalDeserializer()
+	config := kubeletConfig.KubeletConfiguration{}
+	_, _, err = d.Decode(doc, nil, &config)
+	if err != nil {
+		return out, fmt.Errorf("could not decode yaml: %s\n%s", initialConfig, err)
+	}
+
+	// Here we edit the config's fields so we can run the kubelet on Windows
+	config.CgroupDriver = "cgroupfs"
+	config.ResolverConfig = ""
+	cgroupsPerQOS := false
+	config.CgroupsPerQOS = &cgroupsPerQOS
+	config.Authentication.X509.ClientCAFile = filepath.Join(bs.installDir, "kubelet-ca.crt")
+
+	// We need to set EnforceNodeAllocatable with an empty slice, "enforceNodeAllocatable:[]"
+	// the json tags have the field set as `omitempty`, and the field defaults to enforceNodeAllocatable:["pods"]
+	// so we need empty the slice after marshalling
+	// Putting a placeholder value here to be removed later in the function
+	config.EnforceNodeAllocatable = []string{"THIS_MUST_BE_EMPTY"}
+
+	// Turn the config into a json marshalled []byte
+	out, err = json.Marshal(config)
+	if err != nil {
+		return out, err
+	}
+
+	// replacing EnforceNodeAllocatable with an empty slice,
+	outString := strings.Replace(string(out), "\"THIS_MUST_BE_EMPTY\"", "", -1)
+	return []byte(outString), err
+}
+
+// translateFile decodes an ignition "Storage.Files.Contents.Source" field and transforms it via the function provided.
+// if fileTranslateFn is nil, ignitionSource will be decoded, but not transformed
+func (bs *winNodeBootstrapper) translateFile(ignitionSource string, fileTranslateFn translationFunc) ([]byte, error) {
+	contents, err := dataurl.DecodeString(ignitionSource)
+	if err != nil {
+		return []byte{}, err
+	}
+	newContents := contents.Data
+	if fileTranslateFn != nil {
+		newContents, err = fileTranslateFn(bs, contents.Data)
+		if err != nil {
+			return []byte{}, err
+		}
+	}
+	return newContents, err
+}
+
+// These regex are global, so that we only need to compile them once
+// cloudProviderRegex searches for the cloud provider option given to the kubelet
+var cloudProviderRegex = regexp.MustCompile(`--cloud-provider=(\w*)`)
+
+// verbosityRegex searches for the verbosity option given to the kubelet
+var verbosityRegex = regexp.MustCompile(`--v=(\w*)`)
+
+// parseIgnitionFile parses the ignition file and writes the contents of the described files
+// to the k8s installation directory
+func (bs *winNodeBootstrapper) parseIgnitionFile(ignitionFilePath string, filesToTranslate map[string]fileTranslation) error {
```

Comment:
bs seems a bit odd, can we change it to `nbc` or something like that?

Reply:
bs for booststrapper, what would nbc stand for?